pkg/bot/teams.go

```diff
@@ -46,61 +47,59 @@ const (
 
 var _ Bot = &Teams{}
 
+type conversation struct {
+	ref    schema.ConversationReference
+	notify bool
+}
+
 // Teams listens for user's message, execute commands and sends back the response.
 type Teams struct {
 	log             logrus.FieldLogger
 	executorFactory ExecutorFactory
 	reporter        AnalyticsReporter
-	notifyMutex     sync.RWMutex
-	notify          bool
-
-	BotName          string
-	AppID            string
-	AppPassword      string
-	MessagePath      string
-	Port             string
-	AllowKubectl     bool
-	RestrictAccess   bool
-	ClusterName      string
-	Notification     config.Notification
-	Adapter          core.Adapter
-	DefaultNamespace string
-
-	conversationRefMutex sync.RWMutex
-	conversationRef      *schema.ConversationReference
+	// TODO: Be consistent with other communicators when Teams supports multiple channels
+	//channels map[string][ChannelBindingsByName]
+	bindings config.BotBindings
```

Comment:
I cannot see any logic which is setting this field 🤔 The executor would not work properly in this case.